pkg/addons/ensure.go

```diff
@@ -106,20 +115,41 @@ func collectAddons(s *state.State) (addonsToDeploy []addonAction) {
 		})
 	}
 
+	// Special case for AWS
+	if s.Cluster.CloudProvider.AWS != nil {
+		if s.Cluster.CloudProvider.External {
+			addonsToDeploy = append(addonsToDeploy,
+				addonAction{
+					name: resources.AddonCCMAws,
+				},
+				addonAction{
+					name: resources.AddonCSIAwsEBS,
+				},
+			)
+		} else {
+			// since k8s 1.23 CSIMigrationAWS is turn by default and require installation of AWS EBS CSI driver even if
+			// we don't use external ccm (https://github.com/kubernetes/kubernetes/pull/106098)
+			k8sVersion, err := semver.NewVersion(s.Cluster.Versions.Kubernetes)
+			if err != nil {
+				s.Logger.Warnf("Can not check if k8s version '%s' validate constraint '%s' and therefor if CSI driver must be installed. err=%w", s.Cluster.Versions.Kubernetes, greaterThan23Constraint, err)
+			} else if valid, errs := greaterThan23.Validate(k8sVersion); !valid {
+				s.Logger.Infof("K8s version '%s' does not validate constraint '%s'. Using CSI In-Tree implementation . reasons=%s", s.Cluster.Versions.Kubernetes, greaterThan23Constraint, utilerrors.NewAggregate(errs))
```

Comment:
We don't care about this case at all. I'd ignore it and not print anything.

Reply:
done :)